popup.js

```diff
@@ -95,8 +96,11 @@ async function querySecrets(searchString, manualSearch) {
 }
 
 const searchHandler = function (e) {
+  clearTimeout(searchKickoff);
   if (e.key === 'Enter') {
     mainLoaded()
+  } else {
+    searchKickoff = setTimeout(() => mainLoaded(), 1000);
```

Comment:
Yep, that'll give the user two options. Maybe worth experimenting a bit with the timer and see where's the sweet spot. I would perhaps try sub 1s (something in the 700-900 range)